### PR TITLE
Implement file info block which contains information about particular file

### PR DESCRIPTION
**Pull request by AndriiSameliuk**
Implement info block which contains information about particular file.
image

### CODE REVIEW

**Comment by AndriiSameliuk** [comment]
@dkalinin, PTAL.

File: app/View/Component/FileItem/index.js [hunk 1]
```diff
@@ -83,8 +87,17 @@ export class FileItem extends Component {
...
+    this.removeButton.onClick(handler);
+  }
+
+  /**
+   * Sets handler on button if it is clicked.
+   * @public
+   * @param {Function} handler
+   */
+  onInfoButtonClick(handler) {
+    this.infoButton.onClick(handler);
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: this.infoButton.onClick(() => handler(this._id));

File: app/View/Component/FileItem/index.js [hunk 2]
```diff
@@ -181,5 +194,21 @@ export class FileItem extends Component {
...
+    return this._name;
+  }
+
+  /**
+   * Gets type of file.
+   * @return {String}
+   */
+  get type() {
+    return this._type;
+  }
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: Getters are not needed.

File: app/View/Component/FileList/index.js [hunk 3]
```diff
@@ -49,9 +50,30 @@ export class FileList extends StoreAwareComponent {
           this.fileItem = fileItem;
           this.dispatch('renameFile', fileId, name);
         });
+
+        fileItem.onInfoButtonClick(() => {
+          this.fileItem = fileItem;
+          console.log(this.fileItem.id);
+          this.callback(this.fileItem);
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: this.callback(element);

File: app/View/Component/FileList/index.js [hunk 4]
```diff
@@ -49,9 +50,30 @@ export class FileList extends StoreAwareComponent {
...
+  set onFileRemove(callback) {
+    this.removingCallback = callback;
+  }
+
+  /**
+   * Sets callback function on button.
+   * @param {Function} callback
+   */
+  set onInfoButtonClick(callback) {
+    this.callback = callback;
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: Please rename.

File: app/View/Component/FileListPage/index.js [hunk 5]
```diff
@@ -22,7 +23,20 @@ export class FileListPage extends Component {
   initRender() {
     this.div = document.createElement('div');
     this.root.appendChild(this.div);
-    new FileList(this.div);
     this.div.classList.add('drop-zone');
+    const fileList = new FileList(this.div);
+    const hiddenBox = document.createElement('div');
```
Comment (dkalinin) [inline app/View/Component/FileListPage/index.js]: fileInfoBox

File: app/View/Component/FileInfo/index.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  /**
+   * Renders main components.
+   * @private
+   */
+  _render() {
+    this._root.classList.add('opened');
+    this._root.innerHTML = `
+    <span>${this._fileName}</span>`;
```
Comment (dkalinin) [inline app/View/Component/FileInfo/index.js]: Please add more fields.

File: app/View/Component/FileItem/index.js [hunk 7]
```diff
@@ -42,29 +42,32 @@ export class FileItem extends DraggableComponent {
   _render() {
     this._div.innerHTML = ``;
     if (!this.isRenaming) {
-      this.fileName = document.createElement('span');
-      this.fileName.textContent = this.name;
-      this._div.appendChild(this.fileName);
-      this.button = new Button(this._div, 'Delete');
+      const fileName = document.createElement('span');
+      this._fileName = fileName;
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: _renderedFileName

File: app/View/Component/FileList/index.js [hunk 8]
```diff
@@ -34,53 +34,72 @@ export class FileList extends DraggableComponent {
...
+  set onInfoButtonClick(callback) {
+    this.infoButtonCallback = callback;
+  }
 
   /**
    * Checks whether server sends error.
    * @param {String} errorMessage
    */
-  checkServerErrors(errorMessage) {
+  _checkDeletedFile(errorMessage) {
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: Check deleted fie checks whether server sends error?

File: app/View/Component/FileList/index.js [hunk 9]
```diff
@@ -34,53 +34,72 @@ export class FileList extends DraggableComponent {
...
+    this.state.files.forEach((element) => {
+      const fileItem = new FileItem(this._ul, element.name, element.id, element.type);
+      this._fileList.push(fileItem);
 
-        fileItem.onFolderDrop = ((files, folderId) => {
-          this.dispatch('addItems', files, folderId);
-        });
+      fileItem.onRemoveButtonClick(() => {
+        fileItem.disabled = true;
+        this.dispatch('removeItem', element.id);
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: Does it stay disabled if remove request responds with error?

**Comment by AndriiSameliuk** [comment]
@dkalinin, fixed, PTAL.

File: app/View/Component/FileInfo/index.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    <div>${this._fileName}</div>
+    <div>${this._fileOwner}</div>
+    <div>${this._fileUploadDate}</div>`;
+    const closedButton = new Button(this._root, 'Close');
+
+    closedButton.onClick((event) => {
+      event.preventDefault();
+      event.stopPropagation();
+      this._root.classList.remove('opened');
+      this._root.innerHTML = ``;
```
Comment (dkalinin) [inline app/View/Component/FileInfo/index.js]: Please create method _close().

File: app/View/Component/FileInfo/index.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    });
+  }
+  /**
+   * Checks which item deleted if info block of deleted item exist clear it.
+   * @param {String} id
+   */
+  _checkDeletedFile(id) {
+    if (id === this.fileId) {
+      this.root.innerHTML = ``;
```
Comment (dkalinin) [inline app/View/Component/FileInfo/index.js]: Call _close()

File: app/View/Component/FileList/index.js [hunk 12]
```diff
@@ -32,25 +34,48 @@ export class FileList extends DraggableComponent {
...
 
-      this.fileList = [];
-      this.state.files.forEach((element) => {
-        const fileItem = new FileItem(this._ul, element.name, element.id, element.type);
-        this.fileList.push(fileItem);
-        fileItem.onRemoveButtonClick(() => {
-          fileItem.disabled = true;
-          this.dispatch('removeItem', element.id);
-        });
+      this._fileList = [];
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: Please remove.

**Pull request merged**